zone/spells.cpp

```diff
@@ -3977,7 +3975,31 @@ bool Mob::SpellOnTarget(uint16 spell_id, Mob *spelltar, bool reflect, bool use_r
 	cd->hit_heading = action->hit_heading;
 	cd->hit_pitch = action->hit_pitch;
 	cd->damage = 0;
-	if(!IsEffectInSpell(spell_id, SE_BindAffinity) && !IsDamageSpell(spell_id)){
+	
+	// attack.cpp sends out messages to nearby clients for damage and Taps
+	// Here, we only need to send out that type of a message to the caster if we are
+	// the caster or we're the owner of the pet that is the caster.
+	//
+	// This fixes the cases where other clients were getting duplicates of
+	// the messages sent by attack.cpp.
+
+	if (IsDamageSpell(spell_id) || IsLifetapSpell(spell_id)) {
```

Comment:
Tracing the code, I don't think that this section is necessary.

QueueCloseClients should cover sending the packet to the client caster or pet owner as long as they are in range (as defined by DB rule).

It basically sends it to all clients in range

Reply:
But QueueCloseClients  isnt being called anymore due to our change.  These messages disappeared for me with your change alone.

Reply:
In my tests, these messages are gone without my fix.

Reply:
I can give you specific tests if you like.

Reply:
Yes I realized that sorry did not write in here again.

I traced it again, it seems attack.cpp and this are giving us different messages because of the damage value (client perk)

I don't know how it works in live (if both messages get sent, the one that says gets hit by x dmg and the one that says singed for example) but I fixed it in the latest commit